mountpoint-s3-fs/examples/mount_from_config.rs

```diff
@@ -235,14 +247,18 @@ fn main() -> Result<()> {
     // Parse command line arguments
     let args = Args::parse();
     // Read the config
-    let config = load_config(&args.config)?;
+    let config = load_config(&args.config).context("Failed to load config")?;
+    // Set up the event log
+    let event_logger = EventLogger::new(&config.event_log_dir).context("Failed to create an event log")?;
     // Set up logging
-    let (_logging, _metrics) = setup_logging(&config)?;
+    let (_logging, _metrics) = setup_logging(&config).context("Failed to setup logging")?;
     // Process manifests if needed
-    let temporary_dir = tempdir()?;
-    let manifest = process_manifests(&config, temporary_dir.path())?;
+    let temporary_dir =
+        tempdir_in(&config.metadata_store_dir).map_err(|err| anyhow!(err).context("Failed to create manifest"))?;
```

Comment:
`map_err` should not be required. You should be able to call `.context(...)` on any `Error`.